lld/COFF/SymbolTable.cpp

```diff
@@ -56,6 +56,8 @@ static void forceLazy(Symbol *s) {
   }
   case Symbol::Kind::LazyObjectKind: {
     InputFile *file = cast<LazyObject>(s)->file;
+    if (!file->lazy)
```

Comment:
The test is good but this looks like a band aid, which will be unneeded when COFF resolves defined before undefined.

Related, lld/ELF/Symbols.cpp Symbol::extract has an unneeded `file->lazy` check, which I am removing now.

Reply:
Added FIXMEs, thanks!

Reply:
(https://github.com/llvm/llvm-project/commit/c025b96ef9bb364c79f73fc3afb45c851c2efb17 was the mentioned removal in lld/ELF)